app/client/components/contactus/contactUsForm.tsx

```diff
@@ -0,0 +1,291 @@
+import { css, SerializedStyles } from "@emotion/core";
+import { Button } from "@guardian/src-button";
+import { palette, space } from "@guardian/src-foundations";
+import { textSans } from "@guardian/src-foundations/typography";
+import React, { ChangeEvent, FormEvent, useState } from "react";
+import { minWidth } from "../../styles/breakpoints";
+import { Input } from "../input";
+import { ErrorIcon } from "../svgs/errorIcon";
+
+interface ContactUsFormProps {
+  submitCallback: (payload: FormPayload) => void;
+  title: string;
+  subjectLine: string;
+  editableSubjectLine?: boolean;
+  additionalCss?: SerializedStyles;
+}
+
+interface FormPayload {
+  fullName: string;
+  email: string;
+  subjectLine: string;
+  details: string;
+}
+
+interface FormElValidationObject {
+  isValid: boolean;
+  message: string;
+}
+
+interface FormValidationState {
+  inValidationMode: boolean;
+  fullName: FormElValidationObject;
+  email: FormElValidationObject;
+  subjectLine: FormElValidationObject;
+  details: FormElValidationObject;
+}
+
+export const ContactUsForm = (props: ContactUsFormProps) => {
+  const [subjectLine, setSubjectLine] = useState<string>(props.subjectLine);
+  const [fullName, setFullName] = useState<string>(
+    window.guardian?.identityDetails?.displayName || ""
+  );
+  const [email, setEmail] = useState<string>(
+    window.guardian?.identityDetails?.email || ""
+  );
+  const [details, setDetails] = useState<string>("");
+  const [
+    instructionsRemainingCharacters,
+    setInstructionsRemainingCharacters
+  ] = useState<number>(250);
+
+  const mandatoryFieldMessage = "You cannot leave this field empty";
+
+  const [formValidationState, setFormValidationState] = useState<
+    FormValidationState
+  >({
+    inValidationMode: false,
+    fullName: {
+      isValid: true,
+      message: mandatoryFieldMessage
+    },
+    email: {
+      isValid: true,
+      message: mandatoryFieldMessage
+    },
+    subjectLine: {
+      isValid: true,
+      message: mandatoryFieldMessage
+    },
+    details: {
+      isValid: true,
+      message: mandatoryFieldMessage
+    }
+  });
+
+  const validateForm = () => {
+    const isFullNameValid = !!fullName.length;
+    const isEmailValid = !!email.length;
```

Comment:
Should email validation be stronger than just length?

Reply:
we've added some more limited validation to the client